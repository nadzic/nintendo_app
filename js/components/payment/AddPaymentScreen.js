import React, { Component } from 'react';
import {
  Text,
  View,
  StyleSheet,
  TextInput,
  TouchableOpacity,
  Animated,
  Keyboard,
  Platform,
  Image,
} from 'react-native';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import creditCardType from 'credit-card-type';
import { COLORS } from 'nintendoapp/js/constants/colors';
import { closeAddPaymentProblemModal } from 'nintendoapp/js/actions/uiActions';
import ModalSimple from 'nintendoapp/js/components/common/ModalSimple';
import { addCreditCard } from 'nintendoapp/js/actions/paymentActions';
import Loader from 'nintendoapp/js/components/common/Loader';
import NavigationService from 'nintendoapp/js/utils/navigation';
import { formatTextWithRegex } from 'nintendoapp/js/utils/formating';

/* Config/Constants
============================================================================= */

/* eslint-disable global-require */

const FIELDS_WIDTH = '88%';
const BUTTON_HEIGHT = 50;

export class AddPaymentScreen extends Component {
  static navigationOptions = {
    title: 'Add Payment Method',
    headerRight: (<View />),
    headerLeft:
      <TouchableOpacity
        onPress={() => NavigationService.goBack()}
        style={{ marginLeft: 11.5 }}
      >
        <Image
          style={{ width: 22, height: 22 }}
          source={require('nintendoapp/js/components/payment/img/close.png')}
        />
      </TouchableOpacity>,
    headerTitleStyle: {
      fontSize: 20,
      fontWeight: 'normal',
      textAlign: 'center',
      flex: 1,
      fontFamily: 'BrandonText-Medium',
    },
    headerStyle: {
      backgroundColor: COLORS.GREEN,
    },
    headerTintColor: COLORS.WHITE,
  };

  constructor(props) {
    super(props);

    this.keyboardHeight = new Animated.Value(0);

    this.state = {
      ccNumber: '',
      expireDate: '',
      cvvNumber: '',
      zipCode: '',
      ccType: '',
      cvvType: 'CVV',
      cvvLength: 3,
    };
  }

  componentWillMount() {
    if (Platform.OS === 'ios') {
      this.keyboardWillShowSub = Keyboard.addListener('keyboardWillShow', this.keyboardWillShow);
      this.keyboardWillHideSub = Keyboard.addListener('keyboardWillHide', this.keyboardWillHide);
    } else {
      this.keyboardDidShowSub = Keyboard.addListener('keyboardDidShow', this.keyboardWillShow);
      this.keyboardDidHideSub = Keyboard.addListener('keyboardDidHide', this.keyboardWillHide);
    }
  }

  componentWillUnmount() {
    if (Platform.OS === 'ios') {
      this.keyboardWillShowSub.remove();
      this.keyboardWillHideSub.remove();
    } else {
      this.keyboardDidShowSub.remove();
      this.keyboardDidHideSub.remove();
    }
  }

  onSavePress() {
    const {
      ccNumber,
      expireDate,
      zipCode,
      cvvNumber,
    } = this.state;
    const { addCreditCard, user } = this.props;
    const { auth_token } = user;

    const expMonth = expireDate.split('/')[0];
    const expYear = '20'.concat(expireDate.split('/')[1]);

    addCreditCard(auth_token, ccNumber, expMonth, expYear, cvvNumber, zipCode);
  }

  setCcCvvType(ccNumber) {
    const posibleCardsArray = creditCardType(ccNumber);
    if(ccNumber.length <= 4) {
      if (posibleCardsArray.length > 0 && posibleCardsArray.length < 10) {
        this.setState({
          ccType: posibleCardsArray[0].niceType,
          cvvType: posibleCardsArray[0].code.name,
          cvvLength: posibleCardsArray[0].code.size,
        });
      } else {
        this.setState({ ccType: '', cvvType: 'CVV', cvvLength: 3 });
      }
    }
  }

  keyboardWillShow = (event) => {
    if (Platform.OS === 'ios') {
      Animated.timing(this.keyboardHeight, {
        duration: event.duration,
        toValue: event.endCoordinates.height,
      }).start();
    } else {
      this.keyboardHeight = event.endCoordinates.height;
    }
  };

  keyboardWillHide = (event) => {
    if (Platform.OS === 'ios') {
      Animated.timing(this.keyboardHeight, {
        duration: event.duration,
        toValue: 0,
      }).start();
    } else {
      this.keyboardHeight = 0;
    }
  };

  render() {
    const {
      ccNumber,
      expireDate,
      cvvNumber,
      zipCode,
      ccType,
      cvvType,
      cvvLength,
    } = this.state;
    const {
      closeAddPaymentProblemModal,
      loading,
      addPaymentProblemModalOpen,
    } = this.props;
    return (
      <Animated.View style={[styles.container, { paddingBottom: Platform.OS === 'ios' ? this.keyboardHeight : 0 }]}>
        <View style={styles.inputContainer}>
          <View style={styles.firstRow}>
            <TextInput
              style={styles.ccInput}
              placeholder="Card Number"
              underlineColorAndroid="transparent"
              autoCapitalize="none"
              autoCorrect={false}
              keyboardType="numeric"
              autoFocus
              placeholderTextColor={COLORS.LIGHTGREY}
              value={ccNumber}
              onChangeText={(text) => { this.setState({ ccNumber: formatTextWithRegex(text, ' ', '.{1,4}') }); this.setCcCvvType(text); } }
              maxLength={19}
            />
          </View>
          <View style={styles.secondRow}>
            <TextInput
              style={styles.dateInput}
              placeholder="MM/YY"
              underlineColorAndroid="transparent"
              autoCapitalize="none"
              keyboardType="numeric"
              autoCorrect={false}
              placeholderTextColor={COLORS.LIGHTGREY}
              value={expireDate}
              onChangeText={(text) => this.setState({ expireDate: formatTextWithRegex(text, '/', '.{1,2}') })}
              maxLength={5}
            />
            <TextInput
              style={styles.cvvInput}
              placeholder={cvvType}
              underlineColorAndroid="transparent"
              keyboardType="numeric"
              autoCapitalize="none"
              autoCorrect={false}
              placeholderTextColor={COLORS.LIGHTGREY}
              value={cvvNumber}
              onChangeText={(text) => this.setState({ cvvNumber: text })}
              maxLength={cvvLength}
            />
            <TextInput
              style={styles.zipInput}
              placeholder="Zip Code"
              underlineColorAndroid="transparent"
              keyboardType="numeric"
              autoCapitalize="none"
              autoCorrect={false}
              placeholderTextColor={COLORS.LIGHTGREY}
              value={zipCode}
              onChangeText={(text) => this.setState({ zipCode: text })}
            />
          </View>
          <View style={styles.thirdRow}>
            <Text style={styles.ccType}>{ccType}</Text>
          </View>
        </View>
        <TouchableOpacity
          style={styles.saveButton}
          onPress={(e) => this.onSavePress(e)}
        >
          <Text style={styles.saveText}>Save</Text>
        </TouchableOpacity>
        {loading && <Loader />}
        <ModalSimple
          isVisible={addPaymentProblemModalOpen}
          onPress={closeAddPaymentProblemModal}
          headline={`There Was An Issue Adding\nYour Credit Card. Please Try\nAgain Or Use A Different\nCard.`}
        />
      </Animated.View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: COLORS.WHITE,
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'space-between',
  },
  inputContainer: {
    width: FIELDS_WIDTH,
    marginTop: 35,
  },
  firstRow: {
    display: 'flex',
    flexDirection: 'row',
  },
  ccInput: {
    fontFamily: 'BrandonText-Medium',
    flex: 1,
    borderBottomWidth: 1,
    borderBottomColor: COLORS.LIGHTGREY,
    fontSize: 16,
    paddingBottom: 10,
  },
  secondRow: {
    display: 'flex',
    flexDirection: 'row',
    marginTop: 15,
  },
  dateInput: {
    fontFamily: 'BrandonText-Medium',
    flex: 1,
    borderBottomWidth: 1,
    borderBottomColor: COLORS.LIGHTGREY,
    marginRight: 15,
    fontSize: 16,
    paddingBottom: 10,
  },
  cvvInput: {
    fontFamily: 'BrandonText-Medium',
    flex: 1,
    borderBottomWidth: 1,
    borderBottomColor: COLORS.LIGHTGREY,
    marginRight: 15,
    fontSize: 16,
    paddingBottom: 10,
  },
  zipInput: {
    fontFamily: 'BrandonText-Medium',
    flex: 1,
    borderBottomWidth: 1,
    borderBottomColor: COLORS.LIGHTGREY,
    fontSize: 16,
    paddingBottom: 10,
  },
  thirdRow: {
    marginTop: 15,
  },
  ccType: {
    color: COLORS.LIGHTGREY,
  },
  saveButton: {
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: COLORS.GREEN,
    width: FIELDS_WIDTH,
    borderRadius: 5,
    height: BUTTON_HEIGHT,
    marginBottom: 50,
  },
  saveText: {
    fontFamily: 'BrandonText-Medium',
    color: COLORS.WHITE,
    fontSize: 20,
  },
});

const mapStateToProps = (state) =>
  ({
    loading: state.loading,
    user: state.auth.user,
    addPaymentProblemModalOpen: state.ui.addPaymentProblemModalOpen,
  });

const mapDispatchToProps = (dispatch) =>
  bindActionCreators({
    closeAddPaymentProblemModal,
    addCreditCard,
  }, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(AddPaymentScreen);
