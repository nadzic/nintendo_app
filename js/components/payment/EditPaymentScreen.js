import React, { Component } from 'react';
import {
  Text,
  View,
  Image,
  StyleSheet,
  TextInput,
  TouchableOpacity,
  Animated,
  Keyboard,
  Platform,
} from 'react-native';
import { bindActionCreators } from 'redux';
import Modal from 'react-native-modal';
import { connect } from 'react-redux';
import { get as safeGet } from 'lodash';
import { COLORS } from 'nintendoapp/js/constants/colors';
import {
  openRemoveCardModal,
  closeRemoveCardModal,
  closeDeletePaymentProblemModal,
  closeUpdatePaymentProblemModal,
} from 'nintendoapp/js/actions/uiActions';
import { deleteCreditCard, updateCreditCard } from 'nintendoapp/js/actions/paymentActions';
import ModalSimple from 'nintendoapp/js/components/common/ModalSimple';
import NavigationService from 'nintendoapp/js/utils/navigation';
import Loader from 'nintendoapp/js/components/common/Loader';
import { formatTextWithRegex } from 'nintendoapp/js/utils/formating';

/* Config/Constants
============================================================================= */

/* eslint-disable global-require */

const FIELDS_WIDTH = '88%';
const BUTTON_HEIGHT = 50;

export class EditPaymentScreen extends Component {
  static navigationOptions = ({ navigation }) => ({
    title: 'Edit Payment Method',
    headerTitleStyle: {
      fontSize: 20,
      fontWeight: 'normal',
      textAlign: 'center',
      fontFamily: 'BrandonText-Medium',
      flex: 1,
    },
    headerStyle: {
      backgroundColor: COLORS.GREEN,
    },
    headerTintColor: COLORS.WHITE,
    headerRight: (<TouchableOpacity style={{ marginRight: 15 }} onPress={() => navigation.state.params.openRemoveCardModal()}><Image source={require('nintendoapp/js/components/payment/img/delete-white.png')} /></TouchableOpacity>),
    headerLeft:
    <TouchableOpacity
      onPress={() => NavigationService.goBack()}
      style={{ marginLeft: 11.5 }}
    >
      <Image
        style={{ width: 22, height: 22 }}
        source={require('nintendoapp/js/components/payment/img/close.png')}
      />
    </TouchableOpacity>,
  });

  constructor(props) {
    super(props);

    this.keyboardHeight = new Animated.Value(0);

    const creditCardLast4 = safeGet(props.customer, 'sources.data[0].last4', '');
    const expMonth = safeGet(props.customer, 'sources.data[0].exp_month', '');
    const expYear = safeGet(props.customer, 'sources.data[0].exp_year', '');
    const zipCode = safeGet(props.customer, 'sources.data[0].address_zip', '');

    this.state = {
      ccNumber: this.generateCreditCardDisplay(creditCardLast4),
      expireDate: this.generateDateDisplay(expMonth, expYear),
      zipCode,
    };
  }

  componentWillMount() {
    if (Platform.OS === 'ios') {
      this.keyboardWillShowSub = Keyboard.addListener('keyboardWillShow', this.keyboardWillShow);
      this.keyboardWillHideSub = Keyboard.addListener('keyboardWillHide', this.keyboardWillHide);
    } else {
      this.keyboardDidShowSub = Keyboard.addListener('keyboardDidShow', this.keyboardWillShow);
      this.keyboardDidHideSub = Keyboard.addListener('keyboardDidHide', this.keyboardWillHide);
    }
  }

  componentDidMount() {
    if (this.props.openRemoveCardModal && !this.props.navigation.state.params) {
      this.props.navigation.setParams({ openRemoveCardModal: this.props.openRemoveCardModal });
    }
  }

  componentWillUnmount() {
    if (Platform.OS === 'ios') {
      this.keyboardWillShowSub.remove();
      this.keyboardWillHideSub.remove();
    } else {
      this.keyboardDidShowSub.remove();
      this.keyboardDidHideSub.remove();
    }
  }

  onUpdatePress() {
    const {
      expireDate,
      zipCode,
    } = this.state;
    const { updateCreditCard, user } = this.props;
    const { auth_token } = user;

    const expMonth = expireDate.split('/')[0];
    const expYear = '20'.concat(expireDate.split('/')[1]);

    updateCreditCard(auth_token, expMonth, expYear, zipCode);
  }

  confirmRemoveCreditCard = () => {
    const { deleteCreditCard, user } = this.props;
    const { auth_token } = user;

    deleteCreditCard(auth_token);
  }

  generateCreditCardDisplay = (last4) => {
    return last4 ? `**** **** **** ${last4}` : '';
  }

  generateDateDisplay = (expMonth, expYear) => {
    if (expMonth && expYear) {
      if (expMonth.toString().length === 1) {
        expMonth = `0${expMonth}`;
      }
      expYear = expYear.toString().slice(-2);
      return `${expMonth}/${expYear}`;
    }
    return '';
  }

  generateCvvDisplay = (last4) => {
    return last4 ? '***' : '';
  }

  keyboardWillShow = (event) => {
    if (Platform.OS === 'ios') {
      Animated.timing(this.keyboardHeight, {
        duration: event.duration,
        toValue: event.endCoordinates.height,
      }).start();
    } else {
      this.keyboardHeight = event.endCoordinates.height;
    }
  };

  keyboardWillHide = (event) => {
    if (Platform.OS === 'ios') {
      Animated.timing(this.keyboardHeight, {
        duration: event.duration,
        toValue: 0,
      }).start();
    } else {
      this.keyboardHeight = 0;
    }
  };

  render() {
    const {
      closeRemoveCardModal,
      closeDeletePaymentProblemModal,
      closeUpdatePaymentProblemModal,
      loading,
      removeCardModalOpen,
      deletePaymentProblemModalOpen,
      updatePaymentProblemModalOpen,
    } = this.props;
    const {
      ccNumber,
      expireDate,
      zipCode,
    } = this.state;

    return (
      <Animated.View style={[styles.container, { paddingBottom: Platform.OS === 'ios' ? this.keyboardHeight : 0 }]}>
        <View style={styles.inputContainer}>
          <View style={styles.firstRow}>
            <TextInput
              style={styles.ccInput}
              placeholder="Card Number"
              underlineColorAndroid="transparent"
              keyboardType="numeric"
              placeholderTextColor={COLORS.LIGHTGREY}
              value={ccNumber}
              editable={false}
            />
          </View>
          <View style={styles.secondRow}>
            <TextInput
              style={styles.dateInput}
              underlineColorAndroid="transparent"
              placeholder="MM/YY"
              autoCapitalize="none"
              keyboardType="numeric"
              autoCorrect={false}
              placeholderTextColor={COLORS.LIGHTGREY}
              onChangeText={(text) => this.setState({ expireDate: formatTextWithRegex(text, '/', '.{1,2}') })}
              maxLength={5}
              value={expireDate}
            />
            <TextInput
              style={styles.cvvInput}
              placeholder="CVV"
              keyboardType="numeric"
              underlineColorAndroid="transparent"
              placeholderTextColor={COLORS.LIGHTGREY}
              editable={false}
            />
            <TextInput
              style={styles.zipInput}
              placeholder="Zip Code"
              underlineColorAndroid="transparent"
              keyboardType="numeric"
              autoCapitalize="none"
              autoCorrect={false}
              placeholderTextColor={COLORS.LIGHTGREY}
              value={zipCode}
              onChangeText={(text) => this.setState({ zipCode: text })}
            />
          </View>
        </View>
        <TouchableOpacity
          style={styles.updateButton}
          onPress={(e) => this.onUpdatePress(e)}
        >
          <Text style={styles.updateText}>Update</Text>
        </TouchableOpacity>
        <Modal
          isVisible={removeCardModalOpen}
          onBackdropPress={() => closeRemoveCardModal()}
          style={styles.modalContainer}
        >
          <View style={styles.modalMainSection}>
            <Text style={styles.modalHeadlineText}>{`Are You Sure You\nWant To Remove Your\nCard?`}</Text>
            <View style={styles.modalButtonContainer}>
              <TouchableOpacity
                style={styles.modalButtonCancel}
                onPress={() => closeRemoveCardModal()}
              >
                <Text style={styles.modalButtonCancelText}>Cancel</Text>
              </TouchableOpacity>
              <TouchableOpacity
                style={styles.modalButtonYes}
                onPress={() => this.confirmRemoveCreditCard()}
              >
                <Text style={styles.modalButtonYesText}>Yes</Text>
              </TouchableOpacity>
            </View>
          </View>
        </Modal>
        {loading && <Loader />}
        <ModalSimple
          isVisible={deletePaymentProblemModalOpen}
          onPress={closeDeletePaymentProblemModal}
          headline={`There Was An Issue Deleting\nYour Credit Card. Please Try\nAgain.`}
        />
        <ModalSimple
          isVisible={updatePaymentProblemModalOpen}
          onPress={closeUpdatePaymentProblemModal}
          headline={`There Was An Issue Updating\nYour Credit Card. Please Try\nAgain.`}
        />
      </Animated.View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: COLORS.WHITE,
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'space-between',
  },
  inputContainer: {
    width: FIELDS_WIDTH,
    marginTop: 35,
  },
  firstRow: {
    display: 'flex',
    flexDirection: 'row',
  },
  ccInput: {
    flex: 1,
    borderBottomWidth: 1,
    borderBottomColor: COLORS.LIGHTGREY,
    fontSize: 16,
    paddingBottom: 10,
    color: COLORS.LIGHTGREY,
  },
  secondRow: {
    display: 'flex',
    flexDirection: 'row',
    marginTop: 15,
  },
  dateInput: {
    flex: 1,
    borderBottomWidth: 1,
    borderBottomColor: COLORS.LIGHTGREY,
    marginRight: 15,
    fontSize: 16,
    paddingBottom: 10,
  },
  cvvInput: {
    flex: 1,
    borderBottomWidth: 1,
    borderBottomColor: COLORS.LIGHTGREY,
    marginRight: 15,
    fontSize: 16,
    paddingBottom: 10,
  },
  zipInput: {
    flex: 1,
    borderBottomWidth: 1,
    borderBottomColor: COLORS.LIGHTGREY,
    fontSize: 16,
    paddingBottom: 10,
  },
  updateButton: {
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: COLORS.GREEN,
    width: FIELDS_WIDTH,
    borderRadius: 5,
    height: BUTTON_HEIGHT,
    marginBottom: 50,
  },
  updateText: {
    color: COLORS.WHITE,
    fontSize: 20,
  },
  modalContainer: {
    display: 'flex',
    alignItems: 'center',
  },
  modalMainSection: {
    display: 'flex',
    alignItems: 'center',
    width: '100%',
    backgroundColor: COLORS.WHITE,
    borderRadius: 3,
  },
  modalHeadlineText: {
    color: COLORS.BLACK,
    fontSize: 24,
    textAlign: 'center',
    marginTop: 25,
    fontWeight: '500',
  },
  modalButtonContainer: {
    display: 'flex',
    flexDirection: 'row',
  },
  modalButtonCancel: {
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: COLORS.WHITE,
    borderColor: COLORS.GREEN,
    borderWidth: 1,
    width: '44%',
    borderRadius: 5,
    height: BUTTON_HEIGHT,
    marginTop: 25,
    marginBottom: 30,
    marginRight: 8,
  },
  modalButtonCancelText: {
    color: COLORS.GREEN,
    fontSize: 20,
  },
  modalButtonYes: {
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: COLORS.GREEN,
    width: '44%',
    borderRadius: 5,
    height: BUTTON_HEIGHT,
    marginTop: 25,
    marginBottom: 30,
    marginLeft: 8,
  },
  modalButtonYesText: {
    color: COLORS.WHITE,
    fontSize: 20,
  },
});

const mapStateToProps = (state) =>
  ({
    ui: state.ui,
    user: state.auth.user,
    loading: state.loading,
    customer: state.creditCard.customer,
    removeCardModalOpen: state.ui.removeCardModalOpen,
    deletePaymentProblemModalOpen: state.ui.deletePaymentProblemModalOpen,
    updatePaymentProblemModalOpen: state.ui.updatePaymentProblemModalOpen,
  });

const mapDispatchToProps = (dispatch) =>
  bindActionCreators({
    openRemoveCardModal,
    closeRemoveCardModal,
    deleteCreditCard,
    closeDeletePaymentProblemModal,
    closeUpdatePaymentProblemModal,
    updateCreditCard,
  }, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(EditPaymentScreen);
