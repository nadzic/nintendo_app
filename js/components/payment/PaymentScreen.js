import React, { Component, Fragment } from 'react';
import { Text, View, TouchableOpacity, StyleSheet, Image } from 'react-native';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { get as safeGet } from 'lodash';
import { closeProcessingPaymentProblemModal } from 'nintendoapp/js/actions/uiActions';
import { getCreditCard } from 'nintendoapp/js/actions/paymentActions';
import { makeKioskPurchase } from 'nintendoapp/js/actions/cartActions';
import { GO_TO_EDIT_PAYMENT, GO_TO_ADD_PAYMENT, GO_BACK } from 'nintendoapp/js/actions/actionTypes';
import { COLORS } from 'nintendoapp/js/constants/colors';
import { calculateTotal } from 'nintendoapp/js/utils/calculation';
import ModalSimple from 'nintendoapp/js/components/common/ModalSimple';
import Loader from 'nintendoapp/js/components/common/Loader';
import NavigationService from 'nintendoapp/js/utils/navigation';

const FIELDS_WIDTH = '88%';
const FIELDS_HEIGHT = 50;

export class PaymentScreen extends Component {
  static navigationOptions = {
    title: 'Payment',
    headerRight: (<View />),
    headerBackTitle: null,
    headerTitleStyle: {
      fontSize: 20,
      fontWeight: 'normal',
      textAlign: 'center',
      flex: 1,
      fontFamily: 'BrandonText-Medium',
    },
    headerStyle: {
      backgroundColor: COLORS.GREEN,
    },
    headerTintColor: COLORS.WHITE,
    headerLeft: (
      <TouchableOpacity
        style={{ marginLeft: 15 }}
        onPress={() => NavigationService.goBack()}
      >
        <Image
          style={{ width: 30, height: 30 }}
          source={require('nintendoapp/js/components/payment/img/back.png')}
        />
      </TouchableOpacity>),
  }

  componentWillMount() {
    const { getCreditCard, user } = this.props;
    const auth_token = safeGet(user, 'auth_token', '');
    getCreditCard(auth_token);
  }

  onPayClick = () => {
    const {
      makeKioskPurchase,
      user,
      variants,
      quantities,
    } = this.props;
    const { auth_token } = user;
    makeKioskPurchase(auth_token, variants, quantities);
  }

  render() {
    const {
      products,
      quantities,
      closeProcessingPaymentProblemModal,
      customer,
      loading,
      navigation,
      processingPaymentProblemModalOpen,
      goToAddPaymentContentVisible,
    } = this.props;

    const creditCardBrand = safeGet(customer, 'sources.data[0].brand', '');
    const creditCardLast4 = safeGet(customer, 'sources.data[0].last4', '');

    const ccContent = goToAddPaymentContentVisible ? (
      <Fragment>
        <Text style={styles.ccText}>Add payment method</Text>
        <TouchableOpacity onPress={() => navigation.navigate('AddPayment')}>
          <Text style={styles.editText}>Add</Text>
        </TouchableOpacity>
      </Fragment>
    ) : (
      <Fragment>
        <Text style={styles.ccText}>{`${creditCardBrand} ****${creditCardLast4}`}</Text>
        <TouchableOpacity onPress={() => navigation.navigate('EditPayment')}>
          <Text style={styles.editText}>Edit</Text>
        </TouchableOpacity>
      </Fragment>
    );

    return (
      <View style={styles.container}>
        <View style={styles.ccViewer}>
          {ccContent}
        </View>
        <View style={[styles.payButtonSection, goToAddPaymentContentVisible && styles.disabled]}>
          <TouchableOpacity
            style={styles.payButton}
            onPress={() => this.onPayClick()}
            disabled={goToAddPaymentContentVisible}
          >
            <Text style={styles.payText}>Pay <Text style={{ fontFamily: 'BrandonText-Bold' }}>${calculateTotal(products)}</Text></Text>
          </TouchableOpacity>
        </View>
        {loading && <Loader />}
        <ModalSimple
          isVisible={processingPaymentProblemModalOpen}
          onPress={closeProcessingPaymentProblemModal}
          headline={`There Was An Issue\nProcessing Your Payment.\nPlease Try Again Or Add A\nNew Credit Card.`}
        />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignSelf: 'stretch',
    alignItems: 'center',
    backgroundColor: COLORS.WHITE,
  },
  ccViewer: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    backgroundColor: COLORS.LIGHTGREY2,
    width: FIELDS_WIDTH,
    height: FIELDS_HEIGHT,
    borderRadius: 3,
    padding: 15,
    marginTop: 30,
  },
  ccText: {
    fontFamily: 'BrandonText-Bold',
    fontSize: 16,
    fontWeight: 'bold',
  },
  editText: {
    fontFamily: 'BrandonText-Bold',
    color: COLORS.GREEN,
    fontSize: 16,
    fontWeight: 'bold',
  },
  payButtonSection: {
    width: '100%',
    display: 'flex',
    alignItems: 'center',
  },
  payButton: {
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: COLORS.GREEN,
    width: FIELDS_WIDTH,
    borderRadius: 5,
    height: FIELDS_HEIGHT,
    marginTop: 30,
  },
  payText: {
    fontFamily: 'BrandonText-Medium',
    color: COLORS.WHITE,
    fontSize: 20,
  },
  disabled: {
    opacity: 0.3,
  },
});

const mapStateToProps = (state) =>
  ({
    customer: state.creditCard.customer,
    products: state.cart.products,
    variants: state.cart.variants,
    quantities: state.cart.quantities,
    loading: state.loading,
    user: state.auth.user,
    processingPaymentProblemModalOpen: state.ui.processingPaymentProblemModalOpen,
    goToAddPaymentContentVisible: state.ui.goToAddPaymentContentVisible,
  });

const mapDispatchToProps = (dispatch) =>
  bindActionCreators({
    closeProcessingPaymentProblemModal,
    getCreditCard,
    makeKioskPurchase,
  }, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(PaymentScreen);
