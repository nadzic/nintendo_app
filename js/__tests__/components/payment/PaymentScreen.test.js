import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { PaymentScreen } from 'nintendoapp/js/components/payment/PaymentScreen';

Enzyme.configure({ adapter: new Adapter() });

function setup() {
  const props = {
    getCreditCard: jest.fn(),
  };

  const wrapper = shallow(<PaymentScreen {...props} />);

  return {
    props,
    wrapper,
  };
}

describe('PaymentScreen component', () => {
  it('should renders correctly', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });
});
