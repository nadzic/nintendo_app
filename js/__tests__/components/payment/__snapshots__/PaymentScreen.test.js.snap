// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaymentScreen component should renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "stretch",
      "backgroundColor": "#FFF",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#F8F8F8",
        "borderRadius": 3,
        "display": "flex",
        "flexDirection": "row",
        "height": 50,
        "justifyContent": "space-between",
        "marginTop": 30,
        "padding": 15,
        "width": "88%",
      }
    }
  >
    <React.Fragment>
      <Text
        style={
          Object {
            "fontFamily": "BrandonText-Bold",
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
         ****
      </Text>
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
      >
        <Text
          style={
            Object {
              "color": "#0EAF27",
              "fontFamily": "BrandonText-Bold",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          Edit
        </Text>
      </TouchableOpacity>
    </React.Fragment>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "display": "flex",
          "width": "100%",
        },
        undefined,
      ]
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#0EAF27",
          "borderRadius": 5,
          "height": 50,
          "justifyContent": "center",
          "marginTop": 30,
          "width": "88%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFF",
            "fontFamily": "BrandonText-Medium",
            "fontSize": 20,
          }
        }
      >
        Pay 
        <Text
          style={
            Object {
              "fontFamily": "BrandonText-Bold",
            }
          }
        >
          $
          0
        </Text>
      </Text>
    </TouchableOpacity>
  </View>
  <ModalSimple
    headline="There Was An Issue
Processing Your Payment.
Please Try Again Or Add A
New Credit Card."
  />
</View>
`;
